app/api/resources/mentorship_relation.py

```diff
@@ -100,6 +102,9 @@ def put(cls, request_id):
         user_id = get_jwt_identity()
         response = DAO.accept_request(user_id=user_id, request_id=request_id)
 
+        if response[1] is 200:
```

Comment:
Are we sure this index always exists and holds the same expected value everytime? Maybe you can use a `try...except` block here

Reply:
Yes, I think we are sure because even if an error message is returned, it will consist of an error message and a error status code, so there will always be a index at 1 available.

Reply:
i understand the response[1] indexation its not ideal :/ right now i don't have other suggestion or solution to improve this. one of my constant concerns on the backend code is where the error status codes are defined... sometimes on DAO and other times on Resources 🤔 @Hardik0403 @xprilion

Reply:
why not calling this `send_email` function on DAO instead on the resource? Can you do that @Hardik0403 ? In that way you don't have to check here the status code, just call this function after `request.save_to_db()`

Reply:
I too liked your solution and when implemented it, I found it to be more 'bulky' than this solution, as then we would be mocking the mail.send function not just in the DAO file itself, but also in the resources and then in each case in test_dao file since we will be requiring the mocked function to be passed as a parameter to the DAO function, so in my opinion it would just mess up the whole file.
![IMG_20191228_074931](https://user-images.githubusercontent.com/58501114/71537781-fef2dc80-2946-11ea-8b00-0973640d67af.jpg)
Further, this implementation is raising an InterfaceError from sqlite3 as sqlalchemy is taking all these three parameters (including mock send email function) while executing and so it is returning this error. I can't say further much on how this would be working on DAO. 
![IMG_20191228_075132](https://user-images.githubusercontent.com/58501114/71537804-52652a80-2947-11ea-9ad1-550df217a3e5.jpg)


Further I found that the email function previously implemented (sending verification email on registration) was implemented by a similar way on checking the response code. So I feel that this solution may be the go to one. 
![IMG_20191228_074949](https://user-images.githubusercontent.com/58501114/71537779-f69aa180-2946-11ea-8817-7567d9975af0.jpg)


As of now I'm not updating the PR, waiting for further reply.
Thank you:)